src/compiler/parser.ts

```diff
@@ -95,6 +95,979 @@ namespace ts {
         return isMetaProperty(node) && node.keywordToken === SyntaxKind.ImportKeyword && node.name.escapedText === "meta";
     }
 
+    type ForEachChildFunction = <T>(node: any, cbNode: (node: Node) => T | undefined, cbNodes?: (nodes: NodeArray<Node>) => T | undefined) => T | undefined;
+
+    const forEachChildTable = new Array<ForEachChildFunction>(SyntaxKind.Count);

```

Comment:
Some thoughts on this initialization mechanism:
1. AFAIK, the ctor arg does nothing (@jakebailey?)
2. The array will always be sparse (among other things, we're always skipping the first 160 elements)
3. We don't have any protection against specifying two different functions for the same kind.

Reply:
Yes, AFAIK the size argument doesn't actually make this a compact array or anything, hence them trying to add a `Array.withCapacity` instead or something.